### ISSUE TITLE
[Example Request] TransUNet for Medical Image Segmentation on Synapse Dataset

### ISSUE THREAD

**Issue by yassienashrafwasfy** [keras-team/keras-io#2340]
## Summary
Contribute a code example demonstrating TransUNet architecture for multi-organ 
segmentation using the Synapse multi-organ CT dataset.

## Implementation Constraints
- ✅ TensorFlow backend only
- ✅ tf.data API for data pipeline
- ✅ model.fit() for training loop
- ✅ Minimal dependencies only keras + tensorflow + numpy + matplotlib + os
- ✅ Minimal code only 366 lines 
- ✅ Keras Functional API
- ✅ Custom Objects Serialization 

## Why This is a Good Fit for Keras Examples
- Covers an underrepresented domain (medical imaging)
- Demonstrates combining CNNs + Transformers in Keras
- Uses only keras-native training (fit method)

**Comment by innat** [comment]
@yassienashrafwasfy 
You can try creating example using [`medicai`](https://github.com/innat/medic-ai). 

**Some resources:**
- `medicai` code example on [brats-keras-io](https://keras.io/examples/vision/brain_tumor_segmentation/)
- multi-organ synapse dataset (BTCV): `medicai` [code example](https://www.kaggle.com/code/ipythonx/medicai-3d-btcv-segmentation-in-keras/) using **swin-unetr**
- trans-unet model implementation from `medicai` - [documentation](https://github.com/innat/medic-ai/blob/main/medicai/models/transunet/README.md)

**Note:**
Creating a code example for this dataset without using any libraries would be quite challenging, given the need to handle data preprocessing tasks (such as orientation and spacing) as well as implement the core **TransUNet** architecture.

**Comment by yassienashrafwasfy** [comment]
@innat 

- Honestly, great work  you’ve got a lot of models here, and doing both 2D and 3D is really cool.

- I’ve already tried this using 2D slices extracted from 3D scans, and I also used Jaccard and Dice loss. Here are my results. The code isn’t perfect and still needs some refactoring. I’m planning to make it closer to the style of Keras hosted examples. But as a base implementation, it works fine.

<img width="1127" height="414" alt="Image" src="https://github.com/user-attachments/assets/5e0ec65c-6dfd-43dd-981c-ef9fef2bd0c0" />

- I also feel that using a pretrained model from Medici might defeat the purpose of Keras examples, where everything is usually built from scratch. That said, I can reverse engineer parts of it to better understand the details. Would it be okay if I reference or cite parts of your work?

**Comment by dhantule** [comment]
Hi @yassienashrafwasfy, thank you for this example suggestion. 

Please raise a PR if you would like to contribute this example in keras-io, please refer this[ contributor_guide](https://github.com/keras-team/keras-io/blob/master/contributor_guide.md)

**Comment by yassienashrafwasfy** [comment]
@dhantule 
ok i will start working on it
